### ISSUE TITLE
Question: Remote Sessions

### ISSUE THREAD

**Issue by schrotie** [TAAPArthur/MyPersonalXwindowManager#1]
Hi Arthur

I've been carrying around [the idea](https://blog.roggendorf.pro/2019/01/05/scale-me/) of a collaborative online desktop for quite some time. Alas the tech didn't seem to be there quite yet.

What (I think) I need: a (potentially headless) desktop Linux into which multiple users can login via X11 port forwarding (probably wrong tech) or VNC or some such thing. All logged in users shall share the desktop, while each has ones own mouse pointer and keyboard focus.

Could MPXManager be a reasonable part of such a setup?

    Cheers
    Thorsten

**Comment by TAAPArthur** [comment]
@schrotie Yes it could. Assuming you have a X11 server running (headless or otherwise), MPXManger would be work the same as if all users were logged in locally.

Some things to keep in mine:
- Raw X11 forwarding is slow especailly when not on the same lan. [Xpra](https://wiki.archlinux.org/index.php/Xpra#Installation) may be better, but I also haven't tried it in a non-lan setting. Haven't used VNC so don't know how it compares.
- The defining feature of MPXManager is the split focus model. So that multiple people could work on different things on the same machine. It won't, for example, allow 2 peole to seamless edit the same document unless the application is surpising MPX-aware.

- The potential problem of taking a large virtual screen and chunking it up so that it matches a users' physical monitors is already implemented.

- Not sure how many people you plan to support at once. Currently X11 can support ~9 master devices; I have a patch pending to increase it to 64. The number is just hard coded, so it would be trivial to compile your own server that supports an arbitray number of users. Fair resource sharing is another problem.

- While this is a keyboard driven WM, it could be easily setup in a floating layout such that the mouse could be easily used

- A window cannot move between X11 servers. So one couldn't work on a window locally and then send that window to another running server

My interpretation of your blog, is that:

#### Each user would have a workspace(s) where they would do their individual work. 
Audio handling should probably done through puleaudio or such, but you can run a command on workspace switch. The concept of private/or selectivly accessible workspaces could also be implemented. I'm been toying with some ideas but havn't found a usecase as realistic as yours. 

#### Another user would then request for a particular person(s) to look at something.
Would imagine this would be done by sending that window one of their workspaces so they have access to it. There's a built in extension of window-cloning that would give the illusion that both users can access the window. And when finished, the clone could be closed, and the original sent back.

So yes I think this project will be helpful to you; I'm not aware of any other X11 based WM that support a split focus model. If you aren't tied to X11, switching to wayland may open up more options as their seats are different than X11's master devices.
